p2p/discovery/server.go

```diff
@@ -7,20 +7,22 @@ package discovery
 import (
 	"net"
 	"sync"
-	"time"
 
-	"github.com/seeleteam/go-seele/common"
 	"github.com/seeleteam/go-seele/log"
-)
-
-const (
-	PINGPONGINTERVAL  = 10 * time.Second // sleep between ping pong
-	DISCOVERYINTERVAL = 10 * time.Second // sleep between discovery
+	"github.com/seeleteam/go-seele/common/hexutil"
 )
 
 func StartServer(port, id string) {
-	udp := getUDP(port, HexToNodeID(id))
-	log.Debug("nodeid: %s", common.BytesToHex(udp.self.ID.Bytes()))
+	var nodeid NodeID
+	if id == "" {
+		nodeid = getRandomNodeID()
+	} else {
+		nodeid = hexToNodeID(id)
```

Comment:
just curious, how can we ensure the nodeid is a real node?

Reply:
it is passed by command line